From fcb5210d7119c31a858382e0821d99a55f624ea4 Mon Sep 17 00:00:00 2001
From: Mark Hargreaves <clashclanacc2602@gmail.com>
Date: Fri, 27 May 2022 19:34:50 +0300
Subject: [PATCH] make smc draw a stripe, then halt

---
 src/boot/entry.S | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/src/boot/entry.S b/src/boot/entry.S
index f556b36..88f2a32 100644
--- a/src/boot/entry.S
+++ b/src/boot/entry.S
@@ -169,11 +169,21 @@ _exception_vector_el3:
 .balign 128
     b .
 .balign 128
-    /* Lower EL with Aarch64 */
-    mov x18, #0xc
+	ldr x25, =0x87d2e2000   // Framebuffer address
+	mov x26, #0x200000      // Count
+	ldr w27, =0xffdb134b    // Color
+    1:
+        str w27, [x25], #4
+        subs x26, x26, #4
+        bne 1b
+    2:
+    wfi
+    b 2b                    // Loop indefinitely
+
+    mov x18, #0xd           // 0b1101: EL3h
     msr spsr_el3, x18
     eret
-
+    
 .balign 128
     b .
 .balign 128
-- 
2.36.1

